core/store/src/lib.rs

```diff
@@ -285,9 +286,23 @@ pub fn read_with_cache<'a, T: BorshDeserialize + 'a>(
     Ok(None)
 }
 
+lazy_static! {
+    static ref CURRENT_STORE: Mutex<Option<Arc<Store>>> = Mutex::new(None);
```

Comment:
@mikhailOK is there any potential issue with this?

Reply:
this is really ugly, I propose a `CompiledModuleCache` trait and passing a trait object from node-runtime

Reply:
yeah that makes sense

Reply:
Is it really necessary, we only have a singleton store, and it takes 10+ functions to pass store from neard's initialize, which call create_store and have a reference to store to here.

Reply:
> it takes 10+ functions to pass store from neard's initialize

I don't think so. You can pass it from `NightShadeRuntime` in `process_state_update`

Reply:
Ah yes NightShadeRuntime has access to store. but it still takes 10+ function from process_state_update.
```
NightShadeRuntime.process_state_update (last obj has access to store)
runtime.apply
process_receipt
apply_action_receipt
apply_action
action_function_call
execute_function_call
near_vm_runner.run
run_vm
run_wasmer
compile_module_cached
put_module/get_module (need store)
```
I still think pass an argument for these many functions, all of them don't need it at all is more ugly, especially when it's safe singleton. 
put_module and get_module is highly unrelated usual way we used store and rocksdb, i think it can be considered as a separate database, it's just happened to be in same rocksdb, but it's col is isolated and usual store/state operation never reach to CompiledModule column, put_module/get_module doesn't have access to any other column, there is never a race condition

Reply:
We can get Store in `action_function_call` from TrieUpdate, it's not super clean but better than current solution

Reply:
@mikhailOK suggested is a better way, but we have discussed in contract-runtime meeting with @evgenykuzyakov, suggest to introduce Singleton cache, which would also make it relatively easy to disable, we might also get rid of the cached_key! macroses, which are already hard to disable because we need to use compiler flags. We will address cached_key! later;